### ISSUE TITLE
Support HCL

### ISSUE THREAD

**Issue by nikolay** [mikefarah/yq#1844]
**Please describe your feature request.**
HCL is a JSON superset and has a [Go library](https://github.com/hashicorp/hcl) provided by the creator of the format - HashiCorp

**Describe the solution you'd like**
Adding an `hcl` and `h` types in the `-p/--input-format` and `-o/--output-format` options.

**Describe alternatives you've considered**
I've considered using [yj](https://github.com/sclevine/yj) to convert from HCL to JSON, apply [jq](https://github.com/jqlang/jq), and then use `yj` again to convert JSON back to HCL, but it would be nice if `yq` supports HCL natively.

**Additional context**
N/A

**Comment by g13013** [comment]
That would be an awesome addition!

**Comment by nikolay** [comment]
@g13013 I agree. There's no viable alternative at this moment, and Terraform (or OpenTofu), Terramate, and Terragrunt ecosystems will greatly benefit from it.

**Comment by mikefarah** [comment]
v4.50.1 adds support to HCL! Let me know how it does :)
